server/datastore/memorydb/index.ts

```diff
@@ -64,4 +64,14 @@ export class InMemoryDatastore implements Datastore {
     this.posts.splice(index, 1);
     return Promise.resolve();
   }
+
+  getLikes(postId: string): Promise<number> {
+    const likes = this.likes.filter(x => x.postId === postId).length;
+    return Promise.resolve(likes);
+  }
+
+  isDuplicateLike(like: Like): Promise<boolean> {
+     const isExists = this.likes.filter(x => x.postId === like.postId && x.userId === like.userId).length > 0;
```

Comment:
small optimization: you can use `indexOf` instead of `filter` here, since you're really just looking for the first one that matches, there's no need to go through the entire list and test all the likes. :)

Reply:
Good observation.